extractor/src/main.py

```diff
@@ -187,10 +187,20 @@ def broadcast_events(dataframe, analysis, batch_id=None):
                 event["resource_type"] = resource_type
                 event["resource_id"] = resource_id
                 event["record"] = record
-                producer.produce_event(
-                    topic=f"{PRODUCED_TOPIC_PREFIX}{batch_id}",
-                    event=event
-                )
+                try:
+                    producer.produce_event(
+                        topic=f"{PRODUCED_TOPIC_PREFIX}{batch_id}",
+                        event=event
+                    )
+                except KafkaException as e:
+                    if e.args[0].code() == KafkaError.UNKNOWN_TOPIC_OR_PART:
+                        logger.debug(
+                            f"batch {batch_id} may have been deleted: extract topic does not exist"
+                        )
+                        break
```

Comment:
also, we may want to `return` instead of `break` here since we don't want to put the counter in redis if the batch is canceled

Reply:
alright